flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorEventDispatcherImpl.java

```diff
@@ -75,6 +103,36 @@ void dispatchEventToHandlers(
         }
     }
 
+    void initializeOperatorEventGatewayIfExists(StreamOperator<?> operator) throws Exception {
+        if (gatewayMap.containsKey(operator.getOperatorID())) {
+            getOperatorEventGateway(operator.getOperatorID())
+                    .initializeState(getOperatorStateBackend(operator));
+        }
+    }
+
+    void snapshotOperatorEventGatewayIfExists(StreamOperator<?> operator) throws Exception {
+        if (gatewayMap.containsKey(operator.getOperatorID())) {
+            getOperatorEventGateway(operator.getOperatorID())
+                    .snapshotState(getOperatorStateBackend(operator));
+        }
+    }
+
+    void notifyCheckpointAbortedIfExists(StreamOperator<?> operator, long checkpointId) {
+        if (gatewayMap.containsKey(operator.getOperatorID())) {
+            OperatorEventGatewayImpl gateway = getOperatorEventGateway(operator.getOperatorID());
+            gateway.openGateway(checkpointId);
+        }
+    }
+
+    void notifyOperatorSnapshotCreatedIfExists(StreamOperator<?> operator, long checkpointId) {
```

Comment:
Since this method is called right after the snapshot future is created, should it be named `notifyOperatorSnapshotStartedIfExists()`?